### PR TITLE
curl: Add more download and rewrite data examples

### PR DESCRIPTION
**Pull request by ericbn**
- Based on discussion in #1019
- Download: first example introduces `-o`, second one introduces `-O`,
  third example introduces `-L` and `-C` using brackets as per #1018
- Data: first example introduces `-d` (default method is POST and
  default Content-Type is application/x-www-form-urlencoded), second
  example introduces `-X` and `-H`
- Count of examples is still 8 as before

### CODE REVIEW

**Comment by ericbn** [comment]
@agnivade's comments from #1019 suggesting the removal of `--head`. This example exists since the first commit (11264d9).

> Showing the `--head` option is redundant here immediately after showing the `-X METHOD` option. One can easily do a `-X HEAD` to make a head request.
> They are indeed different. `--head` will print only the response headers. This is usually achieved with the HEAD http method. But you can also use the `--head` option with GET method, in which case it ignores the response body and only prints the headers. This is kind of unecessary IMO.
> I would either do `curl -X HEAD http://example.com` or `curl http://example.com -v -o /dev/null`.
> I vote for removing the `--head` example as it can be achieved with the already given options. And given that this page is at `#10` of most searched commands in mixpanel, we should definitely work on making it as simple as possible.

File: pages/common/curl.md [hunk 1]
```diff
@@ -3,34 +3,34 @@
...
 > Supports most protocols including HTTP, FTP, POP.
 
-- Download a URL to a file:
+- Download the contents of an URL to a file:
 
-`curl "{{URL}}" -o {{filename}}`
+`curl {{http://example.com}} -o {{filename}}`
 
-- Send form-encoded data:
+- Download a file saving it under the same name as indicated in the URL:
```
Comment (waldyrious) [inline pages/common/curl.md]: Here the previous description ("Download a file saving the output under the same filename as indicated in the URL") would be more appropriate, IMO, to call attention to the `-O` option, which is different from `-o`.

Maybe we could shorten it a bit to "Download a file saving the output under the filename indicated by the URL", and we might consider writing [O]utput, to match the next examples.

Thoughts?

File: pages/common/curl.md [hunk 2]
```diff
@@ -3,34 +3,34 @@
...
+`curl -O {{http://example.com/filename}}`
 
-- Send JSON data:
+- Download a file, following [L]ocation redirects, and automatically [C]ontinuing/resuming a previous file transfer:
 
-`curl -X POST -H "Content-Type: application/json" -d {{'{"name":"bob"}'}} {{http://localhost/login}}`
+`curl -O -L -C - {{http://example.com/filename}}`
 
-- Specify an HTTP method:
+- POST application/x-www-form-urlencoded data:
```
Comment (waldyrious) [inline pages/common/curl.md]: We should preserve "Send" in the description, for consistency and clarity.

File: pages/common/curl.md [hunk 3]
```diff
@@ -3,34 +3,34 @@
...
+`curl -d {{'name=bob'}} {{http://example.com/form}}`
 
-- Head request:
+- Send data, specifying a custom HTTP method, and including an extra header:
 
-`curl --head {{http://localhost}}`
+`curl -d {{'{"name":"bob"}'}} -X {{PUT}} -H {{'Content-Type: application/json'}} {{http://example.com/users/1234}}`
 
-- Include an extra header:
+- Head request:
```
Comment (waldyrious) [inline pages/common/curl.md]: Let's just remove this one while we're at it, if we all agree it's redundant (I'll defer to @agnivade's comments on this).

File: pages/common/curl.md (hunk 1 above)
Comment (ericbn) [inline pages/common/curl.md]: I was thinking that when there is just one parameter in the example, there should be no need for a `t[e]xt`.

File: pages/common/curl.md (hunk 2 above)
Comment (ericbn) [inline pages/common/curl.md]: I agree, more verbosity here makes is clearer.

File: pages/common/curl.md (hunk 1 above)
Comment (waldyrious) [inline pages/common/curl.md]: Yeah, I agree the [] could be omitted in such cases.

**Comment by ericbn** [comment]
@waldyrious, removed "Head request"  and improved verbosity as discussed in the previous commits. Also changed "POP" to "POP3".

**Comment by agnivade** [comment]
LGTM.

**Comment by waldyrious** [comment]
LGTM as well. Great work and discussion, everyone, and thanks @ericbn for bearing with us :)

**Pull request merged**